### PR TITLE
Adding delete project functionality

### LINKED ISSUE #215: Implement GitProject.delete()
Add method to project classes to delete a project.

AC:
- [ ] In each implementation, add the `delete` method to project class and implement it.
    - If possible/supported. (Document that if it is not possible and raise an exception with a good message.)
- [ ] Add tests for that.

Links:
- https://python-gitlab.readthedocs.io/en/stable/gl_objects/projects.html
- https://pygithub.readthedocs.io/en/latest/github_objects/Repository.html
- https://pagure.io/api/0/ (not found the possible API call for that on the first look)

### PR DESCRIPTION
**Pull request by shreyaspapi**
fixes #215 

- [x] Added tests

### CODE REVIEW

**Comment by shreyaspapi** [comment]
I think I did something wrong here, will take a look.

File: tests/integration/github/test_generic_commands.py [hunk 1]
```diff
@@ -216,3 +216,7 @@ def test_is_private(self):
 
     def test_is_not_private(self):
         assert not self.ogr_project.is_private()
+
+    def test_delete(self):
+        project = self.service.get_project(repo="temp-file", namespace="shreyaspapi")
```
Comment (lachmanfrantisek) [inline tests/integration/github/test_generic_commands.py:221]: Just a nit-pick -- what about using the same name as for others?

**Review by lachmanfrantisek — COMMENTED** [review]
Just one small suggestion. Thanks a lot. It looks so easy and I am wondering why we don't have it already..;)

File: tests/integration/github/test_generic_commands.py (hunk 1 above)
Comment (shreyaspapi) [reply tests/integration/github/test_generic_commands.py]: Done👍

**Review by lachmanfrantisek — APPROVED** [review]
:+1:

**Pull request merged**